### ISSUE TITLE
riff create causes container image name problems on windows

### ISSUE THREAD

**Issue by dmcquiston** [projectriff/riff#568]
Yes, yes, I know, I should be doing this on Linux, but...I figured I'd try the Minikube instructions on Windows using Powershell to see if it can be done. I used the instructions provided here: https://projectriff.io/docs/getting-started-on-minikube/ and actually got pretty far. I was able to install minikube, helm, riff, and the invokers. When I tried to do the first node.js function here was my experience.

I ran this command: 

```batch
riff create node --name square --input numbers --filepath .
```
and received this output:

```batch
Initializing C:\..\square-topics.yaml
Initializing C:\..\square-functions.yaml
Initializing C:\..\Dockerfile
Initializing C:\..\.dockerignore
Building image ...
[STDERR] invalid argument "COMPUTERNAME\\Username/square:0.0.1" for "-t, --tag" flag: invalid reference format: repository name must be lowercase
[STDERR] See 'docker build --help'.
exit status 125
```

Since Create is merely a combination of init, build, and apply, I assume that the build is what failed. So I went ahead and ran each command manually. Init worked the same exact way (see command output above) and was successful. Build failed the same exact way. When I tried a --dry-run it revealed the docker command:

```batch
riff build -n square --dry-run
```

output:
```batch
Building image ...
Build command: docker build -t "COMPUTERNAME\Username/square:0.0.1 .
```

Now, I am a docker n00b, but I don't believe the "COMPUTERNAME\Username" is the right thing to be placed in the beginning part of the tag. I searched high and low and found [this post](https://stackoverflow.com/questions/49357534/invalid-argument-tensorflow-serving-devel-for-t-tag-flaginvalid-referen?rq=1) and it revealed that maybe something else should be here.

FTR, I have run the equivalent of ```eval $(minikube docker-env)``` in Powershell which ran without issue

Any ideas?

**Comment by trisberg** [comment]
Looks like the CLI has a problem looking up your username that is used as the docker account. Try providing it explicitly using `--useraccount` option. I'm labelling this as a CLI bug for now.

**Comment by jldec** [comment]
i can confirm the issue (it's not specific to windows)
the workaround is to override the default user account name with `-u <lowercasename>`

```
~/sq$ riff build -u Hello
Building image ...
[STDERR] invalid argument "Hello/sq:0.0.1" for "-t, --tag" flag: invalid reference format: repository name must be lowercase
[STDERR] See 'docker build --help'.
exit status 125

~/sq$ riff build -u hello
Building image ...
[STDOUT] Sending build context to Docker daemon  6.656kB
[STDOUT] Step 1/3 : FROM projectriff/node-function-invoker:0.0.8
[STDOUT]  ---> 6b42d1262422
[STDOUT] Step 2/3 : ENV FUNCTION_URI /functions/sq.js
[STDOUT]  ---> Using cache
[STDOUT]  ---> 1545eecd63ca
[STDOUT] Step 3/3 : ADD sq.js ${FUNCTION_URI}
[STDOUT]  ---> Using cache
[STDOUT]  ---> 9f424c9b1409
[STDOUT] Successfully built 9f424c9b1409
[STDOUT] Successfully tagged hello/sq:0.0.1
~/sq$ 
```

**Comment by trisberg** [comment]
We still need to handle the `COMPUTERNAME\Username` to strip out the `COMPUTERNAME\` part

**Comment by dmcquiston** [comment]
ok, I got it to work but only after specifying my dockerhub username (which is not your email address, nor, necessarily your Windows Machine name\username.

```batch
riff create node --name square --input numbers --filepath . -u <lowercase-username>
```

I think it would be better to explicitly specify in the [Minikube doc](https://projectriff.io/docs/getting-started-on-minikube/) the ```-u``` parameter indicating what that username.

Thanks for the tip!

**Comment by trisberg** [comment]
Yeah, it's not ideal to default to the current user, although I don't have a better suggestion right now. We do need to point this out in the docs.

You can set an env var named RIFF_USERACCOUNT or create a `.riff.yaml` file in your home directory and set `useraccount: <your-account>` in that file. That way you don't have to provide it every time.

**Comment by dmcquiston** [comment]
I would update the docs if they were on the github somewhere :)

**Comment by scothis** [comment]
@dmcquiston if you're curious https://github.com/projectriff/projectriff.io/blob/master/_docs/010-getting-started-on-minikube.md
